ibis/formats/numpy.py

```diff
@@ -0,0 +1,89 @@
+# move numpy related code here
+import numpy as np
+import toolz
+
+import ibis.expr.datatypes as dt
+import ibis.expr.schema as sch
+
+_from_numpy_types = toolz.keymap(
+    np.dtype,
+    {
+        np.bool_: dt.Boolean,
+        np.int8: dt.Int8,
+        np.int16: dt.Int16,
+        np.int32: dt.Int32,
+        np.int64: dt.Int64,
+        np.uint8: dt.UInt8,
+        np.uint16: dt.UInt16,
+        np.uint32: dt.UInt32,
+        np.uint64: dt.UInt64,
+        np.float16: dt.Float16,
+        np.float32: dt.Float32,
+        np.float64: dt.Float64,
+    },
+)
+
+
+_to_numpy_types = {v: k for k, v in _from_numpy_types.items()}
+
+
+def dtype_from_numpy(typ, nullable=True):
+    if np.issubdtype(typ, np.datetime64):
+        # unit, _ = np.datetime_data(typ)
```

Comment:
This logic enables us proper datetime roundtrips but breaks our testing suite at several places where we expect ns datetime results.